modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/poll/PollingSourceWrapper.java

```diff
@@ -133,7 +137,13 @@ public void onStart(SourceCallback<T, A> sourceCallback) throws MuleException {
         .withName(formatKey("executor")));
 
     stopRequested.set(false);
-    scheduler.schedule(executor, () -> poll(sourceCallback));
+    if (restarting.compareAndSet(true, false)) {
+      delegateRunnable.setDelegate(() -> poll(sourceCallback));
```

Comment:
I don't understand the point of this... if this guy is reconnecting is because it failed while executing. That means that you first simply `poll()` and then simply ` scheduler.schedule(executor, () -> poll(sourceCallback));`... everything else you added seems unnecessary. What am I missing?